source/python/idetect/scraper.py

```diff
@@ -1,18 +1,50 @@
-from flask import Blueprint, render_template
 import newspaper
-import csv
 import urllib
 from urllib import request
 from urllib.parse import urlparse
 import textract
 import os
-from collections import OrderedDict
 import datetime
 from bs4 import BeautifulSoup
 import re
+from sqlalchemy.orm import object_session
+from idetect.model import Content
+from tempfile import NamedTemporaryFile
+
+
+def scrape_article(article):
+    session = object_session(article)
+    # Update the retrieval date and retrieval_attempts
+    article.retrieval_date = datetime.datetime.now()
+    article.retrieval_attempts += 1
+    session.commit()
+    # Attempt to scrape article
+    scrape(article, session)
```

Comment:
Can you call `object_session(article)` in the places within the scrape function that you need the session? That would let you pass just the article in.